src/rest_vol_object.c

```diff
@@ -578,10 +746,368 @@ RV_object_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_object_s
             break;
 
         /* H5Ovisit(_by_name) */
-        case H5VL_OBJECT_VISIT:
-            FUNC_GOTO_ERROR(H5E_OBJECT, H5E_UNSUPPORTED, FAIL,
-                            "H5Ovisit and H5Ovisit_by_name are unsupported");
+        case H5VL_OBJECT_VISIT: {
+            const char *parent_object_header = NULL;
+            iter_data   object_iter_data;
+            hsize_t     idx_p = 0;
+
+            object_iter_data.index_type                   = args->args.visit.idx_type;
+            object_iter_data.iter_order                   = args->args.visit.order;
+            object_iter_data.oinfo_fields                 = args->args.visit.fields;
+            object_iter_data.iter_function.object_iter_op = args->args.visit.op;
+            object_iter_data.op_data                      = args->args.visit.op_data;
+            object_iter_data.is_recursive                 = true;
+            object_iter_data.idx_p                        = &idx_p;
+
+            if (!object_iter_data.iter_function.object_iter_op)
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "no object iteration function specified");
+
+            switch (loc_params->type) {
+                case (H5VL_OBJECT_BY_SELF): {
+
+                    /* Since the VOL doesn't directly pass down the group's hid_t, explicitly open the group
+                     * here so that a valid hid_t can be passed to the user's link iteration callback.
+                     */
+
+                    switch (loc_obj->obj_type) {
+                        case H5I_FILE:
+                        case H5I_GROUP:
+                            if (NULL == (iter_object = RV_group_open(loc_obj, loc_params, ".", H5P_DEFAULT,
```

Comment:
This approach is fine for now since it's what the link code does as well, but eventually it should be switched to use H5VLwrap_register. You can look at the DAOS VOL as an example.